### PR TITLE
Fix doc warnings and errors. Includes a import fix to handle py2.

### PR DESCRIPTION
**Pull request by romeojulietthotel**
The lint.py changes are a result of two things. The import fix was seen when building the docs and the next -> __next__ were seen from 2to3.

### CODE REVIEW

File: werkzeug/contrib/lint.py [hunk 1]
```diff
@@ -126,15 +131,15 @@ class GuardedIterator(object):
 
     def __init__(self, iterator, headers_set, chunks):
         self._iterator = iterator
-        self._next = iter(iterator).next
+        self._next = iter(iterator).__next__
```
Comment (untitaker) [inline werkzeug/contrib/lint.py]: We need to use something like `functools.partial(next, iter(iterator))` here. This has to run under Python 2 as well.

File: werkzeug/contrib/lint.py [hunk 2]
```diff
@@ -126,15 +131,15 @@ class GuardedIterator(object):
...
         self.closed = False
         self.headers_set = headers_set
         self.chunks = chunks
 
     def __iter__(self):
         return self
 
-    def next(self):
+    def __next__(self):
```
Comment (untitaker) [inline werkzeug/contrib/lint.py]: Likewise this doesn't work under Python 2, so additionally you need to apply `implements_iterator` from `werkzeug._contrib` to the entire class.

**Comment by romeojulietthotel** [comment]
I will look at this and make the changes, I was concerned about Python 2 support and my lack of exposure to making things run in both places.

**Comment by untitaker** [comment]
Thanks!

**Pull request merged**